Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -114,6 +116,15 @@ protected ClientBuilderBase()
         {
         }
 
+        /// <summary>
```

Comment:
I think it would be simpler to add a virtual Boolean property that defaults to returning true. We can then use partial classes to disable it for specific APIs.

Reply:
+1, and it would be more in line with this being a builder, which is, this is mutable by defintion.

Not sure wether we want to validat the value being set or not when a credential is explicitly specified. @jskeet what do you think?

Reply:
done